pintc/src/asm_gen/asm_builder.rs

```diff
@@ -757,46 +728,85 @@ impl<'a> AsmBuilder<'a> {
 
                     for word in essential_types::convert::word_4_from_u8_32(predicate_address.1 .0)
                     {
-                        asm.push(ConstraintOp::Stack(Stack::Push(word)));
+                        asm.push(Op::Stack(Stack::Push(word)));
                     }
                 }
             }
 
             ExternalIntrinsic::SizeOf => {
                 // SizeOf is handled separately from other intrinsics, for now.
-                match self.compile_expr_pointer(handler, asm, &args[0], contract, pred)? {
-                    Location::State(next_state) => {
-                        // Remove the placeholder for index computation since it is not needed for
-                        // the `SizeOf` opcode.
-                        asm.push(Stack::Pop.into());
-
-                        asm.push(Stack::Push(next_state as i64).into()); // delta
-                        asm.push(Access::StateLen.into()); // Range length for State
+                // This whole implementation is quite hacky, particularly for `Location::Memory`.
+                // I expect this to change asap.
```

Comment:
This is crap. All of `__size_of` needs to be reworked properly. There's an issue for that.